Shepherd: --merge mode should imply --allow-dirty-main

## Bug

When running `shepherd` with `--merge` / `-m` (full automation mode), unrelated uncommitted changes in the main repo cause the shepherd to exit with code 4 (`NEEDS_INTERVENTION`), requiring the user to manually stash/commit before proceeding.

## Steps to Reproduce

1. Have uncommitted changes in the main repo (e.g., WIP on an unrelated feature)
2. Run: `./.loom/scripts/loom-shepherd.sh 1171 -m`
3. Shepherd exits with code 4:
   ```
   [WARN] Main repository has 3 uncommitted change(s):
     M src/kicad_tools/router/orchestrator.py
     M tests/test_routing_orchestrator.py
     M uv.lock
   [ERROR] Main repo has uncommitted changes that could cause test inconsistencies.
   ```

## Expected Behavior

`--merge` mode implies the user wants fully automated end-to-end orchestration. It should also imply `--allow-dirty-main`, since:

1. The builder works in an isolated worktree anyway — uncommitted main changes don't affect it
2. The user explicitly opted into autonomous mode, so blocking for manual intervention contradicts the intent
3. The workaround (stash, run shepherd, pop stash) is tedious and error-prone

## Proposed Fix

In `cli.py`, when `--merge` is set, also set `allow_dirty_main = True`. The check can still **warn** about uncommitted changes (useful for debugging), but shouldn't **block** in merge mode.

```python
# In main() or argument resolution:
if args.force_merge:
    args.allow_dirty_main = True
```

## Location

- `loom-tools/src/loom_tools/shepherd/cli.py` — `_check_main_repo_clean()` (lines ~223-270)
- `loom-tools/src/loom_tools/shepherd/cli.py` — `main()` where the check is called (lines ~1896-1897)

--- a/loom-tools/src/loom_tools/shepherd/cli.py
+++ b/loom-tools/src/loom_tools/shepherd/cli.py
@@ -1890,6 +1890,12 @@ def main(argv: list[str] | None = None) -> int:
     repo_root = find_repo_root()
     _auto_navigate_out_of_worktree(repo_root)
 
+    # --force / --merge implies --allow-dirty-main: the user wants fully
+    # autonomous operation and the builder works in an isolated worktree,
+    # so uncommitted main changes shouldn't block.
+    if args.force:
+        args.allow_dirty_main = True
+
     # Pre-flight check: warn if main repo has uncommitted changes.
     # This prevents confusion when tests pass in main but fail in worktrees
     # (or vice versa) due to uncommitted local changes.

